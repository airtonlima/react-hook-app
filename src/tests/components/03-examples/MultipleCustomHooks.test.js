import React from 'react';
import { shallow } from "enzyme";
import { MultipleCustomHooks }  from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en <MultipleCustomHooks />', () => {
       
    useCounter.mockReturnValue({ 
        state: 10, 
        increment: () => {}
    })

    test('#01 - Debe de mostrarse correctamente', () => {
        
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })
        
        const wrapper = shallow( <MultipleCustomHooks /> )
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('#02 - ', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Airton',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        })
        
        const wrapper = shallow( <MultipleCustomHooks/> )
        
        expect( wrapper.find('.alert').exists() ).toBe(false)
        expect( wrapper.find('.mb-10').text().trim() ).toBe('Hola Mundo')
        expect( wrapper.find('footer').text().trim() ).toBe('Airton')

        // console.log( wrapper.html() )
    })
    
});