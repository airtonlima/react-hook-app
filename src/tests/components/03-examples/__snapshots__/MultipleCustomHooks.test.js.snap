// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> #01 - Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <br />
  <div
    className="alert alert-info text-center"
  >
    Cargando ...
  </div>
  <button
    className="btn btn-primary m-10"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</Fragment>
`;
